### PR TITLE
Issue 3336 - suggest config and cert version be indicated on initial …

### PR DESCRIPTION
**Pull request by LiilyZhang**
…agent install

Signed-off-by: zhangl <zhangl@us.ibm.com>

# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Additional Context (Please include any Screenshots/gifs if relevant)

...

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have tagged the reviewers in a comment below incase my pull request is ready for a review
- [x] I have signed the commit message to agree to Developer Certificate of Origin (DCO) (to certify that you wrote or otherwise have the right to submit your contribution to the project.) by adding "--signoff" to my git commit command.

### CODE REVIEW

File: cutil/cutil.go [hunk 1]
```diff
@@ -796,3 +797,30 @@ func GetDockerEndpoint() string {
...
+// extrace the version in certificate file. The version should be in the first line in this format -----x.x.x-----
+func GetCertificateVersion(certFilePath string) string {
+	var firstLine string
+	var version string
+
+	glog.Info("try another way to read")
+	if cert, err := os.Open(certFilePath); err != nil {
+		glog.Info(fmt.Sprintf("unable to open Certificate file %v, error %v", certFilePath, err))
+	} else if certBytes, err := ioutil.ReadAll(cert); err != nil {
+		glog.Info(fmt.Sprintf("unable to read ESS SSL Certificate file %v, error %v", certFilePath, err))
```
Comment (linggao) [inline cutil/cutil.go:810]: This is the cert file for talking to the exchange. Please remove the work ESS.

File: cutil/cutil.go [hunk 2]
```diff
@@ -796,3 +797,30 @@ func GetDockerEndpoint() string {
...
+	} else {
+		lines := strings.Split(string(certBytes), "\n")
+		if len(lines) > 0 {
+			firstLine = lines[0]
+		}
+		if firstLine != "" {
+			parts := strings.Split(firstLine, "-----")
+			if len(parts) == 3 {
+				version = parts[1]
```
Comment (linggao) [inline cutil/cutil.go:819]: What will be returned if the cert file does not contain the version?

Comment (LiilyZhang) [reply cutil/cutil.go:818]: @linggao If cert file doesn't have version, the version returned will be an empty string. 
The `exchangesync/node.go` will not get certificate version. Then the local db and exchange node will show empty string from cert version

Comment (LiilyZhang) [reply cutil/cutil.go:818]: @linggao I got you. Yes, you are right. Will add check for that. I think I added that check in `agent-install.sh`, need to check if extract `version != BEGIN CERTIFICATE`

Comment (linggao) [reply cutil/cutil.go:818]: for old cert file that starts with "-----BEGIN CERTIFICATE-----", the function will return "BEGIN CERTIFICATE" as the version. Is it right?

Comment (LiilyZhang) [reply cutil/cutil.go:818]: @linggao yes, will fix it

**Review by dlarson04 — APPROVED** [review]

**Pull request merged**